### DISCUSSION TITLE
virtualuser table available (so your login is not public)?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [processone/ejabberd#4292] [Questions]
I was wondering if there is something like a virtualuser table like mta are having. Currently I am using prosody and ldap auth and I don't want to have user login accounts made public.

Eg. so you can have a public xmpp address thisismyemail@example.com which is then mapped to userloginaccount@example.com or even userloginaccount@chat.example.com? Sort of like a virtual user table in a MTA

**Comment by badlop** [comment]
Or, in other words, register an account, and then allow the user to register an "account name alias": another account that would forward all traffic to and from the main account? Such a thing is not implemented.


I guess in your user database you keep track of all that information:
```
{userloginaccount, password, real name, telephone number, thisismyemail}
```
If you want your users to have as JID `thisismyemail@example.com`, then configure the server to use that field as username, and tell your users to authenticate in the XMPP server using `thisismyemail@example.com`.

By the way, as an alternative to the native LDAP authentication support, you could write an external authentication script in any programming language that you want, that performs any operations that you would like:

https://docs.ejabberd.im/developer/guide/#external-authentication
https://docs.ejabberd.im/admin/configuration/authentication/#external-script

**Reply by f1-outsourcing** [reply]
> Or, in other words, register an account, and then allow the user to register an "account name alias": another account that would forward all traffic to and from the main account? Such a thing is not implemented.
> 

No publish qwerqwerwqer@example.com and use this, but authenticate with useraccount@chat.example.com, useraccount@chat.example.com should never become public.

> I guess in your user database you keep track of all that information:
> 
> ```
> {userloginaccount, password, real name, telephone number, thisismyemail}
> ```
> 

not really relevant.

> If you want your users to have as JID `thisismyemail@example.com`, then configure the server to use that field as username, and tell your users to authenticate in the XMPP server using `thisismyemail@example.com`.
> 

no I want to authenticate with useraccount. The whole idea is not to publish your login!

> By the way, as an alternative to the native LDAP authentication support, you could write an external authentication script in any programming language that you want, that performs any operations that you would like:
> 
> https://docs.ejabberd.im/developer/guide/#external-authentication https://docs.ejabberd.im/admin/configuration/authentication/#external-script

I am not sure yet if you can solve this at authentication

**Reply by licaon-kter** [reply]
>  The whole idea is not to publish your login!

publish where? what does this mean?

**Reply by f1-outsourcing** [reply]
everyone knows that if your email address is abuse@gmail.com, then you login in on the webinterface of gmail.com with abuse@gmail.com
I don't want to do that my xmpp address should be different from my login.
